src/messages.rs

```diff
@@ -581,6 +611,20 @@ pub enum MessageContent {
         /// The `part_index`-th part of the serialised user message.
         payload: Vec<u8>,
     },
+    /// Send among Group Y to vote for Accept or Reject a joining node
+    CandidateApproval(bool),
+    /// Approves the joining node as a routing node.
+    ///
+    /// Sent from Group Y to the joining node.
+    NodeApproval {
+        /// The routing table shared by the nodes in our group, including the `PublicId`s of our
+        /// contacts.
+        sections: Vec<(Prefix<XorName>, Vec<PublicId>)>,
```

Comment:
Let's make this a `BTreeMap<Prefix<XorName>, BTreeSet<PublicId>>`, so we don't have to sort manually.
(I'll define this type as `SectionMap` in https://github.com/maidsafe/routing/pull/1315.)